Teensy/src/gpio/gpio.cpp

```diff
@@ -94,9 +95,19 @@ static GyroData_s gpio_getGyro()
     return gyro;
 }
 
+// defining resistors for voltage divider
+constexpr float BATTERY_SCALAR = (4000.0f + 1000.0f ) / 1000.0f;
+// the ADC analog input is returned in 10 bit format 
+#define ADC_MAX_VALUE 1023
+// the ADC can only handle 3.3V
+#define ADC_MAX_VOLTAGE 3.3
 static daq_BatteryV_t gpio_getVBat()
 {
-    return 0U;
+    float batteryAnalogIn = analogRead(BATTERY_SENSOR_PIN);
+    // the formula for 2 resistors connected in series, forming the voltage divider is: V1 = Vm * (R2/(R1+R2))
+    float batteryVoltage_dV = ((batteryAnalogIn / ADC_MAX_VALUE) * ADC_MAX_VOLTAGE) * BATTERY_SCALAR;
```

Comment:
```suggestion
    daq_BatteryV_t batteryVoltage_dV = (daq_BatteryV_t) ((batteryAnalogIn / ADC_MAX_VALUE) * ADC_MAX_VOLTAGE) * BATTERY_SCALAR * 10;
```
Sorry my fault, forgot to add times 10 as we want the value in deci-volts

Reply:
got it, sorry, forgot about that as well as I thought the data was originally in deci-volts to begin with.